### ISSUE TITLE
False positive for `PT005` and function returning `Iterator[None]`

### ISSUE THREAD

**Issue by ZeeD** [astral-sh/ruff#12742]
Minimal snippet:

```python
from collections.abc import Iterator

import pytest

@pytest.fixture()
def _my_fixture_none() -> None:
  return

@pytest.fixture()
def _my_fixture_itor() -> Iterator[None]:
  yield

def test_mytest(
  _my_fixture_none: None,
  _my_fixture_itor: None,
):
    ...
```

ruff returns error `PT005` on `_my_fixture_itor`. It shouldn't.

According to https://docs.pytest.org/en/latest/reference/reference.html#pytest-fixture 

> Fixtures can provide their values to test functions using return or yield statements. When using yield the code block after the yield statement is executed as teardown code regardless of the test outcome, and must yield exactly once.

So a fixture that work only by side effect may `return` `None` or `Iterator[None]` - and in fact if you use a fixture that `return`s `Iterator[None]` have the variable set to `None`

**Comment by charliermarsh** [comment]
Is the suggestion that both implicit `return` and implicit `yield` (without a value) should be allowed?

**Comment by charliermarsh** [comment]
I don't get a PT005 error on this: https://play.ruff.rs/b404e34f-2c6e-4228-a300-cd783adb9980. Can you help me reproduce?

**Comment by ZeeD** [comment]
Sorry, it seems there was a fluke from my side.

I got the `PT005` because in my actual code I have explicit `yield None`
In fact I see that I can see the `PT005` if I write

```python
@pytest.fixture
def _my_fixture_none() -> None:
    return None

@pytest.fixture
def _my_fixture_itor() -> Iterator[None]:
    try:
        yield None
    finally:
        ...
```

cfr: https://play.ruff.rs/cd4496f2-4176-4a30-8471-8dee0b8d090a


regarding this gh issue, I understand that the explicit `None` is not necessary, but should be "functionally equivalent" of the bare `return` and `yield`, and it seems a bit strange to me how this is related to the `PT005` itself. In any case I can see that the error was in my code, not in ruff.

Maybe it could be possible to better rephrase the error message / add a note in the documentation page?

**Comment by MichaReiser** [comment]
Thanks for the example! 

We decided to deprecate `PT005` because it has other flaws and isn't recommended by the community. I close the issue because we don't plan on making improvements to the rule further.
